source/src/AnalysisHelper.cc

```diff
@@ -0,0 +1,149 @@
+/// \file AnalysisHelper.cc
+/*
+ *
+ * AnalysisHelper.cc source template automatically generated by a class generator
+ * Creation date : mar. juil. 1 2014
+ *
+ * This file is part of DQM4HEP libraries.
+ *
+ * DQM4HEP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * based upon these libraries are permitted. Any copy of these libraries
+ * must include this copyright notice.
+ *
+ * DQM4HEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with DQM4HEP.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Tom Coates
+ * @copyright CNRS , IPNL
+ */
+
+// -- dqm4hep headers
+#include <dqm4hep/AnalysisHelper.h>
+
+namespace dqm4hep {
+
+  namespace core {
+
+    //----------------------------------------------------------------------------------------------------
+
+    float AnalysisHelper::mainHelper(MonitorElementPtr pMonitorElement, std::string testType, float percentage)
+    {
+
+      if (nullptr == pMonitorElement) {
+        throw StatusCodeException(STATUS_CODE_INVALID_PTR);
+      }
+
+      TH1 *h = pMonitorElement->objectTo<TH1>();
```

Comment:
The `AnalysisHelper::mean` function should also work for other object such as TGraph (on the y axis)